src/dstack/_internal/core/backends/oci/compute.py

```diff
@@ -73,12 +106,77 @@ def run_job(
         project_ssh_public_key: str,
         project_ssh_private_key: str,
     ) -> JobProvisioningData:
-        raise NotImplementedError
+        instance_config = InstanceConfiguration(
+            project_name=run.project_name,
+            instance_name=get_instance_name(run, job),
+            ssh_keys=[SSHKey(public=project_ssh_public_key.strip())],
+            job_docker_config=None,
+            user=run.user,
+        )
+        return self.create_instance(instance_offer, instance_config)
 
     def terminate_instance(
         self, instance_id: str, region: str, backend_data: Optional[str] = None
     ) -> None:
-        raise NotImplementedError
+        region_client = self.regions[region]
+        region_client.compute_client.terminate_instance(
```

Comment:
Can this raise an error if instance_id not found? dstack may terminate an already deleted instance, and terminate_instance() should not raise errors in such case.